pipeline/ingestor/ingestor_common/src/main/scala/weco/pipeline/ingestor/common/IngestorMain.scala

```diff
@@ -0,0 +1,67 @@
+package weco.pipeline.ingestor.common
+
+import com.sksamuel.elastic4s.Index
+import io.circe.{Decoder, Encoder}
+import weco.elasticsearch.IndexConfig
+import weco.elasticsearch.typesafe.ElasticBuilder
+import weco.messaging.typesafe.SNSBuilder
+import weco.pipeline_storage.Indexable
+import weco.pipeline_storage.elastic.{ElasticIndexer, ElasticSourceRetriever}
+import weco.pipeline_storage.typesafe.PipelineStorageStreamBuilder
+import weco.typesafe.WellcomeTypesafeApp
+import weco.typesafe.config.builders.AkkaBuilder
+import weco.typesafe.config.builders.EnrichConfig._
+
+import scala.concurrent.ExecutionContext
+
+trait IngestorMain[In, Out] extends WellcomeTypesafeApp {
```

Comment:
why does this need to be a trait and not a class with a contructor? it doesn't look like we override the behaviour in its usages

Reply:
I think you want to extend this in the `Main` object, no? I'm not sure what this would look like with a class.

Reply:
hmm I think you could avoid extending `WellcomeTypesafeApp` here and keep that duplicated and just have a class for everything else that is duplicated between the 2 ingestors no?
I think there is more to be gained avoiding too many overrides than by removing absolutely all code duplication, especially in a class that is actually never tested. It should be as readable as possible and overrides make everything harder to follow for me.